src/main/kotlin/org/jetbrains/research/kotlin/mpp/inference/graph/NodeIO.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.research.kotlin.mpp.inference.graph
+
+import org.jetbrains.research.kotlin.mpp.inference.tensors.Tensor
+
+@Suppress("UNCHECKED_CAST")
+class NodeIO(names: List<String> = emptyList()) : LinkedHashMap<String, Tensor<*>?>() {
```

Comment:
Wow, do you *really* need to inherit from `LinkedHashSet`? It is super strange, better just composition